### DISCUSSION TITLE
Question: possible to use micronaut-serialization library alone (without the micronaut framework)?

### DISCUSSION THREAD

**Discussion by dxxvi** [micronaut-projects/micronaut-serialization#912] [Q&A]
### Issue description

This is the only micronaut-related library I use:

```
    <dependency>
      <groupId>io.micronaut.serde</groupId>
      <artifactId>micronaut-serde-jackson</artifactId>
      <version>2.10.2</version>
    </dependency>
```

Could somebody write me some code to serialize an instance of this class?

```java
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public class GameScore {
  @JsonProperty(value = "game_id", required = true)
  private final String gameId; // playerId is in the message key

  @JsonProperty(value = "score", required = true)
  private final int score;

  @JsonCreator
  public GameScore(String gameId, int score) {
    this.gameId = gameId;
    this.score = score;
  }

  public String getGameId() {
    return gameId;
  }

  public int getScore() {
    return score;
  }
}
```

**Comment by graemerocher** [comment]
maven or Gradle?

**Reply by hrgdavor** [reply]
maven. I am also looking into using annotation processing to pre-compile serialization and deserialization code instead of runtime.

**Reply by hrgdavor** [reply]
I tried adding the processor into maven build

```xml
<plugin>
	<artifactId>maven-compiler-plugin</artifactId>
	<configuration>
		<proc>full</proc>
		<source>21</source>
		<target>21</target>
		<annotationProcessorPaths>
			<annotationProcessorPath>
				<groupId>io.micronaut.serde</groupId>
				<artifactId>micronaut-serde-processor</artifactId>
				<version>2.15.1</version>
			</annotationProcessorPath>
		</annotationProcessorPaths>
	</configuration>
</plugin>

```

I marked one record with `@Serdeable` but nothing was generated

I also could not find direct processor name to use with `org.bsc.maven:maven-processor-plugin`

**Reply by graemerocher** [reply]
you need `micronaut-inject-java` as well

**Reply by hrgdavor** [reply]
Hi, thanks for looking into this. Can you provide more details please :)..


I added these dependencies directly in pom:

```xml
<dependencies>
	<dependency>
		<groupId>io.micronaut.serde</groupId>
		<artifactId>micronaut-serde-jackson</artifactId>
		<version>2.15.1</version>
	</dependency>
	<dependency>
		<groupId>io.micronaut.serde</groupId>
		<artifactId>micronaut-serde-processor</artifactId>
		<version>2.15.1</version>
		<scope>provided</scope>
	</dependency>
	<dependency>
		<groupId>io.micronaut</groupId>
		<artifactId>micronaut-inject-java</artifactId>
		<version>4.9.10</version>
		<scope>provided</scope>
	</dependency>
</dependencies>
```

then I added these to processorPaths

```xml
<annotationProcessorPaths>
	<annotationProcessorPath>
		<groupId>io.micronaut.serde</groupId>
		<artifactId>micronaut-serde-processor</artifactId>
		<version>2.15.1</version>
	</annotationProcessorPath>
	<annotationProcessorPath>
		<groupId>io.micronaut</groupId>
		<artifactId>micronaut-inject-java</artifactId>
		<version>4.9.10</version>
	</annotationProcessorPath>
</annotationProcessorPaths>
```

I placed @Serdeable on one record that is read from yml config file using snakeyaml and jackson.

I did not get anything generated and have no errors or warning when compiling

**Comment by hrgdavor** [comment]
I am still stuck with maven setup mentioned above.

@graemerocher I may be barking at the wrong tree here. I prefer more standalone libraries, even from micronaut that is leaner than Spring (that is my impression at least).

I do not want to bother you much if micronaut team has no time or desire to make parts more standalone. It is fine to do so, I just need to know to evaluate my options. 

I am looking into https://avaje.io/jsonb/ next, even though I use jackson for all of my projects for a long time now, and would prefer to find something standalone(not too many deps) that generates code for serialization/deserialization for jackson.
